lib/smb.c

```diff
@@ -472,17 +538,20 @@ static CURLcode smb_connect(struct Curl_easy *data, bool *done)
   return CURLE_OK;
 }
 
-static CURLcode smb_recv_message(struct Curl_easy *data, void **msg)
+static CURLcode smb_recv_message(struct Curl_easy *data,
+                                 struct smb_conn *smbc,
+                                 void **msg)
 {
-  struct connectdata *conn = data->conn;
-  struct smb_conn *smbc = &conn->proto.smbc;
-  char *buf = smbc->recv_buf;
+  char *buf;
   ssize_t bytes_read;
   size_t nbt_size;
   size_t msg_size;
-  size_t len = MAX_MESSAGE_SIZE - smbc->got;
+  size_t len;
   CURLcode result;
 
+  buf = smbc->recv_buf;
+  len = MAX_MESSAGE_SIZE - smbc->got;
```

Comment:
Is there a rationale to move these initializations away from the declarations?

Reply:
@Spiffyk no, that is a leftover from having had a `smbc == NULL` in an interim version that was removed later.

Reply:
Just pushed the revert of that now unnecessary change.